File: src/main/java/br/com/me/backendchallenge/domain/Item.java
package br.com.me.backendchallenge.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.math.BigDecimal;

@Getter
@Setter
@Entity
@Table(name = "TB_ITEM")
@AllArgsConstructor
@NoArgsConstructor
public class Item {
    @Id
    @Column(name = "ID")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JsonIgnore
    @ManyToOne
    @JoinColumn
    private Pedido pedido;

    private String descricao;
    private BigDecimal precoUnitario;
    private Long qtd;
}

File: src/MercadoEletronico.Teste.Domain/Entities/Item.cs
using System;

namespace MercadoEletronico.Teste.Domain.Entities
{
    public class Item : Entity
    {
        public Guid PedidoId { get; set; }
        public string Descricao { get; set; }
        public int PrecoUnitario { get; set; }
        public int Quantidade { get; set; }
        public Pedido Pedido { get; set; }
    }
}